dotnet-interactive.IntegrationTests/KernelSmokeTests.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.DotNet.Interactive.App.IntegrationTests.Utility;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.DotNet.Interactive.App.IntegrationTests
+{
+    public class KernelSmokeTests
+    {
+        private async Task SendSubmitCodeToProcess(Process process, string code)
+        {
+            var submissionJson = $@"{{""token"":""abc"", ""commandType"":""SubmitCode"", ""command"":{{""code"":""{code}"", ""submissionType"":0, ""targetKernelName"":null}}}}";
+            await process.StandardInput.WriteLineAsync(submissionJson);
+            await process.StandardInput.FlushAsync();
+        }
+
+        private async Task<bool> WaitForCondition(Func<bool> predicate, TimeSpan timeout)
+        {
+            var startTime = DateTime.UtcNow;
+            while (DateTime.UtcNow < startTime + timeout)
+            {
+                if (predicate())
+                {
+                    return true;
+                }
+
+                await Task.Delay(200);
+            }
+
+            return predicate();
+        }
+
+        [IntegrationTheory]
+        [InlineData("csharp")]
+        [InlineData("fsharp")]
+        public async Task smoke_test_kernels(string kernelName)
+        {
+            var outputMessages = new List<JObject>();
+            var errorLines = new List<string>();
+            var process = ProcessHelper.StartInteractiveServer(
+                kernelName: kernelName,
+                output: line => outputMessages.Add(JObject.Parse(line)),
+                error: errorLines.Add);
+
+            await SendSubmitCodeToProcess(process, "1+1");
+
+            // waiting for something like this:
+            // {
+            //   "event": {
+            //     "value": 2,
+            //     ...
+            //   },
+            //   "eventType": "ReturnValueProduced",
+            //   ...
+            // }
+
+            var foundIt = await WaitForCondition(() =>
+            {
+                foreach (var obj in outputMessages)
+                {
+                    if (obj["eventType"]?.ToString() == "ReturnValueProduced" &&
+                        obj["event"]["value"]?.ToString() == "2")
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }, TimeSpan.FromSeconds(5));
+
+            foundIt.Should().BeTrue("Unable to find expected result in:\n" + string.Join("\n", outputMessages.Select(o => o.ToString())));
+            errorLines.Should().BeEmpty();
```

Comment:
We have `ToSubscribedList` for gathering up events, and the event definitions under `Microsoft.DotNet.Interactive.Events` are intended to be serialized, with public serialization APIs (instead of `JObject`). If we use those APIs we'll get better maintainability (if the contract changes, for example) and better test feedback. One example where the current test can fail without being informative (which is all the more important in a test that's typically running only on the server): 

* `foundIt` is false but we can't see why.
* We don't know if `errorLines` was empty, and we can't see the errors that were returned.